aten/src/ATen/NamedTensor.cpp

```diff
@@ -1,14 +1,37 @@
 #ifdef NAMEDTENSOR_ENABLED
 #include <ATen/NamedTensor.h>
+#include <ATen/core/Tensor.h>
+#include <torch/csrc/utils/memory.h>
 
 namespace at {
 
 bool NamedTensorMeta::has_names() const {
   return !std::all_of(
-      names.begin(), names.end(), [](const Dimname& n) {
+      names_.begin(), names_.end(), [](const Dimname& n) {
```

Comment:
Are there cases where this would be called many times repeatedly on the same tensor without the names having changed? Any thoughts on if it could cause performance issues or if we would need to cache the result?

Reply:
Oh we could cache this. I don't want to write in too many optimizations right now in fear that they might be premature, so I'll leave that for the future.